dql/parser.go

```diff
@@ -1721,8 +1721,15 @@ L:
 							itemInFunc.Errorf("len function only allowed inside inequality" +
 								" function")
 					}
-					function.Attr = nestedFunc.NeedsVar[0].Name
-					function.IsLenVar = true
+					if len(function.Attr) == 0 {
+						// eq(len(a), 1)
+						function.Attr = nestedFunc.NeedsVar[0].Name
+						function.IsLenVar = true
+					} else {
+						// eq(1, len(a))
```

Comment:
This doesn't seem wrong from the look of it. Is there a reason we shouldn't support this case?

Reply:
We don't support this in case of `eq(Age, val(A))`, we silently fail in that case. If we want to add support for symmetric equality check, we should begin at the more common examples. The way code is structured, it would require more effort than the scope of this here.